### DISCUSSION TITLE
Deployment error: Failed to collect page data, next/dist/build/utils.js

### DISCUSSION THREAD

**Discussion by AndrewSB** [vercel/vercel#8654] [Help] [answered]
hey! i'm not sure what i've done, but it looks like `next build`'s telemetry step, after building, is failing on vercel's servers... the process exits successfully on my machine, so i'm kind of in the dark on how to debug this

this is from attempting to deploy `master` on https://github.com/AndrewSB/andrw-wave. i successfully deploying a few commits ago, at https://github.com/AndrewSB/andrw-wave/commit/b11d927450b12eaf6d757e6b5a2bd5b41a667272

here are the build logs from the vercel CI, it looks like `/vercel/path0/node_modules/next/dist/build/utils.js:916:15` is failing with an `Error'`

```
Detected Next.js version: 12.3.1
--
03:30:38.943 | Running "npm run build --verbose"
03:30:39.219 | npm verb cli /node12/bin/node /node16/bin-npm7/npm
03:30:39.219 | npm info using npm@8.11.0
03:30:39.219 | npm info using node@v12.22.11
03:30:39.219 | npm timing npm:load:whichnode Completed in 0ms
03:30:39.220 | npm timing config:load:defaults Completed in 1ms
03:30:39.220 | npm timing config:load:file:/node16/lib/node_modules/npm/npmrc Completed in 1ms
03:30:39.220 | npm timing config:load:builtin Completed in 1ms
03:30:39.220 | npm timing config:load:cli Completed in 2ms
03:30:39.220 | npm timing config:load:env Completed in 0ms
03:30:39.220 | npm timing config:load:file:/vercel/path0/.npmrc Completed in 0ms
03:30:39.220 | npm timing config:load:project Completed in 1ms
03:30:39.220 | npm timing config:load:file:/vercel/.npmrc Completed in 0ms
03:30:39.220 | npm timing config:load:user Completed in 0ms
03:30:39.221 | npm timing config:load:file:/node12/etc/npmrc Completed in 0ms
03:30:39.221 | npm timing config:load:global Completed in 0ms
03:30:39.221 | npm timing config:load:validate Completed in 0ms
03:30:39.221 | npm timing config:load:credentials Completed in 1ms
03:30:39.221 | npm timing config:load:setEnvs Completed in 1ms
03:30:39.221 | npm timing config:load Completed in 7ms
03:30:39.221 | npm timing npm:load:configload Completed in 7ms
03:30:39.221 | npm timing npm:load:mkdirpcache Completed in 0ms
03:30:39.221 | npm timing npm:load:mkdirplogs Completed in 1ms
03:30:39.221 | npm verb title npm run build
03:30:39.222 | npm verb argv "run" "build" "--loglevel" "verbose"
03:30:39.222 | npm timing npm:load:setTitle Completed in 1ms
03:30:39.222 | npm timing config:load:flatten Completed in 2ms
03:30:39.222 | npm timing npm:load:display Completed in 4ms
03:30:39.222 | npm verb logfile logs-max:10 dir:/vercel/.npm/_logs
03:30:39.225 | npm verb logfile /vercel/.npm/_logs/2022-09-29T10_30_39_204Z-debug-0.log
03:30:39.225 | npm timing npm:load:logFile Completed in 5ms
03:30:39.225 | npm timing npm:load:timers Completed in 0ms
03:30:39.225 | npm timing npm:load:configScope Completed in 0ms
03:30:39.225 | npm timing npm:load Completed in 19ms
03:30:39.231 |  
03:30:39.231 | > andrw-wave@0.1.0 build
03:30:39.231 | > next build
03:30:39.231 |  
03:30:39.681 | Attention: Next.js now collects completely anonymous telemetry regarding usage.
03:30:39.681 | This information is used to shape Next.js' roadmap and prioritize features.
03:30:39.682 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
03:30:39.682 | https://nextjs.org/telemetry
03:30:39.682 |  
03:30:39.786 | info  - Linting and checking validity of types...
03:30:43.632 | info  - Creating an optimized production build...
03:30:52.972 | info  - Compiled successfully
03:30:52.972 | info  - Collecting page data...
03:30:53.319 | SyntaxError: Unexpected token '.'
03:30:53.319 | at Loader.moduleStrategy (internal/modules/esm/translators.js:140:18)
03:30:53.319 | at async link (internal/modules/esm/module_job.js:42:21)
03:30:53.320 |  
03:30:53.321 | > Build error occurred
03:30:53.323 | Error: Failed to collect page data for /
03:30:53.323 | at /vercel/path0/node_modules/next/dist/build/utils.js:916:15
03:30:53.324 | at processTicksAndRejections (internal/process/task_queues.js:97:5) {
03:30:53.324 | type: 'Error'
03:30:53.324 | }
03:30:53.350 | npm timing command:run Completed in 14121ms
03:30:53.352 | npm verb exit 1
03:30:53.352 | npm timing npm Completed in 14146ms
03:30:53.352 | npm verb code 1
03:30:53.375 | Error: Command "npm run build --verbose" exited with 1
```

any thoughts on what could be wrong? or how i can go deeper into debugging why the build crashes on vercel's CI?

**Comment by amyegan** [comment, marked as answer]
_tl;dr The solution was to update the project to Node 16_

---

Hi @AndrewSB! I think there is a conflict in your dependencies. Here's what I see when I pull down the latest `master` commit and try to `npm install`:

```
❯ npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: react-typed@1.2.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR!   peer react@">=16.8" from @react-hook/latest@1.0.3
npm ERR!   node_modules/@react-hook/latest
npm ERR!     @react-hook/latest@"^1.0.2" from @react-hook/resize-observer@1.2.6
npm ERR!     node_modules/@react-hook/resize-observer
npm ERR!       @react-hook/resize-observer@"^1.2.6" from the root project
npm ERR!   9 more (@react-hook/passive-layout-effect, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.3.0" from react-typed@1.2.0
npm ERR! node_modules/react-typed
npm ERR!   react-typed@"^1.2.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^16.3.0" from react-typed@1.2.0
npm ERR!   node_modules/react-typed
npm ERR!     react-typed@"^1.2.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

You probably already have copies of the older dependencies locally from previous iterations, which would explain why you don't have this problem. And might explain why your local `next build` is working but is failing the CI build.

I was able to work around the installation error by using `--legacy-peer-deps`. This also updated `package-lock.json`. After that, I was able to use `npm run build` and deploy to Vercel without seeing any errors. I was not able to reproduce the error you saw.

While I don't think that Next.js telemetry is the source of the problem, you can [opt out](https://nextjs.org/telemetry#how-do-i-opt-out) by running `next telemetry disable` in the root directory of your project. If it was the cause, you should be able to get a successful build after that.

**Reply by AndrewSB** [reply]
hey Amy! i really appreciate you taking such a close look

ahh yeah, i left that part out because i thought it wasn't relevant. for more detail, before i tried building locally, i:
1. deleted `node_modules`
2. deleted `.next/`
3. deleted `.vercel/`
4. ran `npm i --legacy-peer-deps`

and still got a successful build on my machine

"Telemetry" might be the wrong word, i think more accurately the step that's failing is where `next` calculates which pages are static/server generated/etc.

you can see, that the compile finished successfully, and then the `Collecting page data` fails. `Collecting page data` succeeding looks like this:

<img width="1045" alt="Screen Shot 2022-09-29 at 10 40 31 AM" src="https://user-images.githubusercontent.com/3814772/193105658-494bd9ca-b0ad-4f14-967c-76f0de8c5140.png">

is there some way i can get a more specific error message for you to share with the team? i think this is actually an unhandled crash inside next's build tool 

i found https://vercel.com/guides/deploying-locally-built-nextjs, attempting this for now to unblock my release, but i'd love to have the underlying issue fixed so next time i work on this i can use the cloud build again

**Reply by amyegan** [reply]
I'm glad you have a workaround to unblock your deployment for now. Do you have any Build & Development Settings overrides? And does it make a difference in your build result if you override the Install Command to be `npm install --legacy-peer-deps`?

**Reply by AndrewSB** [reply]
i figured it out, i was running one more deployment and i saw the warning about Node v12 becoming deprecated next week

**Reply by AndrewSB** [reply]
I updated to Node v16 in the vercel dashboard, and the builds started succeeding. maybe let your team know that the latest version of the vercel tool crashes under some conditions on Node 12.x? or not, it seems so low priority since its EOLing next week

**Reply by amyegan** [reply]
I'm glad you figured out what was causing it. Thanks for coming back here to share the solution!
